Miscellaneous minor changes

LukeShu opened the pull request:
There's no over-arching theme to this PR, it's just a set of several minor improvements that I've ended up with, and figured I should push upstream.

Commit:
alpm-hook.sh: simplify

Instead of using a temporary file, use process substitution.  This
results in (IMO) clearer code; you can see how the different parts
relate.  It also means you don't have to worry about leaking temporary
files, which probably isn't an issue, but makes me nervous.

majewsky commented on doc/holo-files.8.pod:
```diff
@@ -98,6 +98,14 @@ This means that when configuration is applied as a holoscript on top of the
+recognizes C<arch>, C<debian>, C<fedora>, C<suse>, and (for testing purposes)
+C<unittest>.  If for some reason L<os-release(5)> can't be relied upon, then it
+may be overridden by setting the C<$HOLO_FILES_CURRENT_DISTRIBUTION> environment
+variable to one of the recognized distributions.
```
Please keep `unittest` and `$HOLO_FILES_CURRENT_DISTRIBUTION` out of the official documentation. Both are strictly testing instruments. If `os-release(5)` cannot be relied upon, users should be pushed into submitting bug reports to their distribution (or to us) instead of monkeying around with their environment.

majewsky commented on doc/holo-plugin-interface.7.pod:
```diff
@@ -265,9 +265,9 @@ If the entity does not have a meaningful diff (e.g. for the C<run-scripts>
-applied (or force-applied) by the plugin, the second file represents the current
-state of the entity.
+descriptor 3. The first file represents the state of the entity as it was last
+applied by the plugin (ie, the state the plugin expects it to currently be in),
```
`s/ie/i.e./` (or "that is")

LukeShu replied:
*Id est*, but the lack of periods is IMO far from the most egregious mutilations of punctuation with common Latin abbreviations.

majewsky commented on src/distribution-integration/alpm-hook.sh:
```diff
@@ -2,12 +2,9 @@
+       sort | sed 's+^+/+') \
+     <(# enumerate files that are managed by holo-files
+       holo scan --short | sed -n 's/^file://p') \
+| sed 's/^/file:/' | xargs -r holo apply
```
Since this is not covered by the test suite: Did you test this change?

LukeShu replied:
I did.  But now that you say that, I'm figuring out how to add it to the test suite.

In doing so, I realized that it doesn't work correctly with HOLO_ROOT_DIR.  I guess I'll fix that.  But it's being highlighted to me that holo-files embeds HOLO_ROOT_DIR into the entity IDs, which seems gross to me.

Commit:
test: Add test for alpm-hook.sh

LukeShu commented:
I believe I've addressed your concerns.

User-facing changes:
 - holo-files now looks for os-release(5) in `HOLO_ROOT_DIR`, rather than the actual root directory.
 - pacman integration should now work with `pacman --root /some/chroot`.

majewsky commented on src/distribution-integration/alpm-hook.sh:
```diff
@@ -2,12 +2,13 @@
-rm /tmp/files-managed-by-holo
+# ALPM calls this script from the root directory, with respect to the
+# `--root` flag.
+export HOLO_ROOT_DIR="."
```
This cannot be left in here. Any non-empty `HOLO_ROOT_DIR`will force plugins into testing mode.

I can certainly see why you did this, though. In testing mode, entity IDs look like `file:etc/foo/bar`, while actual entity IDs in production look like `file:/etc/foo/bar`, with the trailing slash in front. That's a divergence that we should rather address in Holo itself, so that it always uses absolute paths in entity IDs, whether in testing mode or not.

Since the rest of the changes looks good, I will cherry-pick all your commits except for the ones regarding alpm-hook into master now. Please rebase after that.

LukeShu replied:
> Any non-empty `HOLO_ROOT_DIR` will force plugins into testing mode.

Is that actually a problem (will one of them do something silly during `scan`)?  The hook will only apply `files:` entities, and at this point `holo-files` has *no* testing mode.

And support for `pacman --root` was kinda cool.

majewsky replied:
I agree about `pacman --root`. You learn something new every day.

Re testing mode: It's true that `holo-files` has no testing mode *at the moment*, but it could gain one, and then things will be weird. I've been bitten by "ah, that edge case will never happen in production" a few times too often in my (as of yet short) career.
